docs/server/clusters.mdx

```diff
@@ -1,15 +1,114 @@
+</Tip>
+
+
+## Deploying Restate clusters
+
+To deploy a distributed Restate cluster without external dependencies, you need to configure the following settings in your [server configuration](/server/configuration):
+
+```toml restate.toml
+# Every node needs to have a unique node name
+node-name = "UNIQUE_NODE_NAME"
+# All nodes need to have the same cluster name
+cluster-name = "CLUSTER_NAME"
+# Make sure it does not conflict with the other nodes
+advertised-address = "ADVERTISED_ADDRESS"
+# At most one node can be configured with auto-provision = true
+auto-provision = false
+# Default replication factor for both the logs and the partitions.
+#
+# Replicate the data to 2 nodes. This requires that the cluster has at least 2 nodes to
+# become operational. If the cluster has at least 3 nodes, then it can tolerate 1 node failure.
+#
+# This also controls the default partition replication. A value of 2 means each partition
+# will be running on (max) of 2 nodes, ensuring redundancy and fault tolerance.
+default-replication = 2
+
+[bifrost]
+# Only the replicated Bifrost provider can be used in a distributed deployment
+default-provider = "replicated"
+
+[metadata-server]
+# To tolerate node failures, use the replicated metadata server
+type = "replicated"
+
+[metadata-client]
+# List all the advertised addresses of the nodes that run the metadata-server role
+addresses = ["ADVERTISED_ADDRESS", "ADVERTISED_ADDRESS_NODE_X"]
+
+[admin]
+# Make sure it does not conflict with the other nodes
+bind-address = "ADMIN_BIND_ADDRESS"
+
+[ingress]
+# Make sure it does not conflict with other nodes
+bind-address = "INGRESS_BIND_ADDRESS"
+```
+
+It is important that every Restate node you start has a **unique `node-name`** specified.
+This defaults to the hostname. When restate starts, it will look in the `restate-data` directory for a folder with its node name as name, and use that data.
+For example, if node A tried to start on a new node with node_name `node-a` and it finds a `restate-data` directory containing data for `node-b`, then this situation will lead to data loss!
+So, in principle, nodes can have identical configurations except for the node name.
+
+All nodes that are part of the cluster need to have the same `cluster-name` specified.
+
+At most one node can be configured with `auto-provision = true`.
+If no node is allowed to auto provision, then you have to manually provision the cluster.
+Refer to the [Cluster provisioning](#cluster-provisioning) section for more information.
+
+The log provider needs to be configured with `default-provider = "replicated"`.
+The `default-replication` should be set to the number of nodes that the data should be replicated to.
+If you run at least `2 * default-replication-property - 1` nodes, then the cluster can tolerate `default-replication-property - 1` node failures.
+
+The metadata server type should be set to `replicated` to tolerate node failures.
```

Comment:
This has been the default for a little while; it's only necessary to set this if using the `object-store` metadata backend with Amazon S3.

```suggestion
Metadata availability is crucial for cluster availability, and the default metadata server type `replicated` can tolerate node failures.
```